README_EN.md

```diff
@@ -0,0 +1,240 @@
+FRDIntent/URLRoutes supports simple url parameter pattern match. In the example above, we register url with the pattern `"/story/:storyId"`, if there is an external call like `frdintent://frdintent.com/story/123`, FRDIntent/URLRoutes will save the key `storyId` and the value `123` into the block handler's parameter `params` that is a dictionary. We can get `123` from the dictionary `params` by key `storyId`.
+
+
+## Attention
+
+#### Prefix
+
+Prefix is not necessary in Swift project because of Swift's visiblity declaration. But considering that this library will be used widely in Objective-C project, it's better to add `FRD` prefix to avoid the name conflict in Objective-C.
+
+#### Type of `source` parameter
+
+In the class `FRDControllerManager`, the methods `startController(source: UIVieController, intent: FRDIntent)` and `startControllerForResult(source: UIViewController, intent: FRDIntent, requestCode: Int)` don't constrain the type of `source` parameter. The precise type of `source` is like what does `UIViewController<FRDIntentReceivable>` or `UIViewController<FRDIntentForResultReceivable>` present: it's a specific class conforming a specific protocol. This must be implemented awkwardly by generic type in Swift 3. The generic type is not precise in this situation. And it's more terrrible that the generic method in Swift cann't be used by Objective-C. So we do a tradeoff to declare the source just as UIViewController type. The user of this library must ensure that the source conform the `FRDIntentReceivable` protocol or the `FRDIntentForResultReceivable` protocol by himself.
+
+
+## Intent 和 URLRoutes
+
+FRDIntent/Intent and FRDIntent/URLRoutes can cooperate with each other. FRIntent/Intent handles internal call and FRDIntent/URLRoutes handles external call. In the implementation of FRDIntent/URLRoutes, FRDIntent/URLRoutes only exposes the external call's entry. In the app, FRDIntent/Intent effectively handles the launch process for FRDIntent/URLRoutes.
+
+The separation of FRDIntent/Intent and FRDIntent/URLRoutes is effectively the separation of internal call and external call. There are some advantages of this separation:
+
+- FRDIntent/Intent is better for internal call. It supports to transfer complex data and customize transition animation. Those are hard to be realized in the URL Router's solution.
+- It's better to separate internal call and external call. We have the opportunity to choose which internal call should be exposed to external. In the URL Router's solution, it's impossible to do this separation. All registers are exposed to both internal and external. We haven't the opportunity to choose.
+
+
+## FRDIntentDemo
+
+FRDIntent is implemented in Swift. FRDIntentDemo is implemented in Objective-C for show its good Objective-C compatiblity.
+
+For test external call, input `frdintent://frdintent.com/user/123` in Safari's address text field. If everything is normal, FRDIntentDemo will be launched and display FirstViewController.
```

Comment:
To test external calls, enter `frdintent://frdintent.com/user/123` in Safari's address bar. If everything goes fine, FRDIntentDemo will launch and display the FirstViewController.